### ISSUE TITLE
Allow custom functions for sanitizing some struct fields

### ISSUE THREAD

**Issue by Daksh14** [dashxhq/sanitizer#4]
Example

```rust
#[derive(Sanitize)]
struct Test {
  #[sanitize(custom(sanitize_string))]
  field: String
}
fn sanitize_string(field: String) -> String {
  // sanitize field here
  field
}

fn main() {
 let instance: Test = Default::default();
 instance.sanitize()?;
}
```
A custom function to sanitize a field will be called in the sanitize trait itself. The signature should match the type of the field. Rust will take care of the return type so we won't have to worry about that.

**Comment by RabidFire** [comment]
I like this. It's quite similar to the validator crate too. We would need it to work for Int types & String types.

**Comment by Daksh14** [comment]
Closed by e25ee38eba3b5be71260de9217eba9b7092150c7
